python/ray/dag/dag_node_operation.py

```diff
@@ -32,13 +47,27 @@ def __init__(
                 as bind_index because there may be more tasks bound to an actor
                 than tasks that appear in the current compiled DAG.
             operation_type: The type of operation to perform.
+            method_name: The name of the method that this operation originates
+                from. This is only for debugging purposes.
         """
         self.exec_task_idx = exec_task_idx
         self.type = operation_type
+        self.method_name = method_name
 
     def __repr__(self):
         return f"(Task idx: {self.exec_task_idx}, Type: {self.type})"
 
+    def __str__(self):
+        return f"([{self.exec_task_idx}] {self.method_name} {self.type})"
+
+    def __hash__(self):
```

Comment:
I think I had it in earlier version of the code. Adding `__hash__()` does not hurt and is convenient so I left it. Let me know if you think otherwise, I can remove it.